lib/LedgerSMB/Sysconfig.pm

```diff
@@ -495,7 +499,7 @@ sub check_permissions {
 
     my $tempdir = LedgerSMB::Sysconfig::tempdir();
     # commit 6978b88 added this line to resolve issues if HOME isn't set
-    $ENV{HOME} = $tempdir;
+    local $ENV{HOME} = $tempdir;
```

Comment:
That is in the queue.

We really should not be using `%ENV` as we are.   The idea of pushing our config variables into `%ENV` is just bad.

When you see a `$LSMB::Sysconfig::variable` you know it was processed--it is ours.  That leaves `%ENV`  just to be `local`ized and initialized in small call chains for "export" to subshells.
That means ignoring `%ENV` other than at startup.

For example LSMB_ALWAYS_MONEY is set many places and it is deleted in App_State, IIRC.
The flow is unknown to me, but what would happen if it were true at start up.  It is not vetted in Sysconfig and there are probably no tests.

`%ENV{HOME}` is set in Sysconfig.  Why? That is rhetorical, but it is hard to answer unless you were there.  `%ENV{HOME}` only appears in Sysconfig.  If it were brought into `$LSMB::Sysconfig::home`  it would self document when pushed out again.   

The enviros LSMB_ALWAYS_MONEY, LSMB_NOHEAD, REQUEST_URI are not handled by
Sysconfig.  That is not bad, but having a convention where enviros are generally set, makes the question require an answer.  When the answer is just that it was missed, it just reinforces my point that the work of forcing the environment is a lot of work for no gain.

The short version:  global variables are bad.